builder/proxmox/iso/config.go

```diff
@@ -68,6 +78,90 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, []string, error) {
 		errs = packersdk.MultiErrorAppend(errs, errors.New("when specifying iso_url, iso_storage_pool must also be specified"))
 	}
 
+	// each device type has a maximum number of devices that can be attached.
+	// count iso, disks, additional isos configured for each device type, error if too many.
+	ideCount := 0
+	sataCount := 0
+	scsiCount := 0
+	virtIOCount := 0
+
+	if c.ISODevice == "" {
+		// set default ISO boot device ide2 if no value specified
+		log.Printf("iso_device not set, using default 'ide2'")
+		c.ISODevice = "ide2"
+		ideCount++
+		// Pass c.ISODevice value over to common config to ensure device index not used by disks
+		c.ISOBuilderCDROMDevice = "ide2"
+	} else {
+		// Pass c.ISODevice value over to common config to ensure device index not used by disks
+		c.ISOBuilderCDROMDevice = c.ISODevice
+		// get device from ISODevice config
+		rd := regexp.MustCompile(`\D+`)
+		device := rd.FindString(c.ISODevice)
+		// get index from ISODevice config
+		rb := regexp.MustCompile(`\d+`)
+		_, err := strconv.Atoi(rb.FindString(c.ISODevice))
+		if err != nil {
+			errs = packersdk.MultiErrorAppend(errs, errors.New("iso_device value doesn't contain a valid index number. Expected format is <device type><index number>, eg. scsi0. received value: "+c.ISODevice))
+		}
+		// count iso
+		switch device {
+		case "ide":
+			ideCount++
+		case "sata":
+			sataCount++
+		case "scsi":
+			scsiCount++
+		}
+		for idx, iso := range c.AdditionalISOFiles {
+			if iso.Device == c.ISODevice {
```

Comment:
Does that `iso.Device` come from proxmox, or is it coming from the SDK? Wondering if this should also be updated to only specify the device type so the plugin can handle the numbering automatically
Of course if this comes from the SDK and we already support it this way, we cannot remove that capability on a minor/patch version, please let me know if this is the case, and we can ignore/revisit my suggestion then.

Reply:
iso.Device here is the Device field for an additional_iso_files block (from common's Config struct). This was added to ensure the ISO device in the iso builder's Config struct doesn't conflict with a device assignment in the common builder's Config struct, an example where this could happen from a user's perspective would be having `iso_device = sata0` and `additional_iso_files{ device = sata0 }` configured at the same time.

Reply:
Gotcha, so this is indeed part of the proxmox plugin, so we can change this for this plugin. I would maybe revisit the way we add those iso devices to the VM config, using a single function from `common`, this way we can use it from both the `Clone` and `ISO` builders, and not have this logic scattered around.
Maybe we can add this as a step is relevant? Otherwise a function should do the trick possibly.
Let me know what you think about this idea, and if you need some help with the code, feel free to ask and I can take a jab at that!

Reply:
With the background obtained answering the above comment on device indexes, I think iso and additional_iso_files could potentially be merged together and maybe the packer config adapted to just have 1 or more `iso` blocks rather than `iso` fields/values outside of a block for the first ISO then multiple `additional_iso_files` blocks for each after? They could be processed in the order they're defined as the disks blocks are, that way you could consistently define the first ISO as an Operating System installer media, second is an autounattend.xml for Windows builds (just examples here), third VirtIO drivers and so on. 
I do then wonder what the point of the iso builder would be if all the ISO logic moves to common.